src/Resources/app/administration/src/module/findologic-module/components/findologic-config/index.js

```diff
@@ -54,19 +54,25 @@ Component.register('findologic-config', {
   data () {
     return {
       isLoading: false,
-      term: null,
-      categories: [],
-      categoryIds: []
+      categoryCollection: null,
     };
   },
 
   created () {
-    this.createdComponent();
+    this.createCategoryCollection();
   },
 
   methods: {
-    createdComponent () {
-      this.getCategories();
+    async createCategoryCollection () {
+      this.categoryCollection = this.actualConfigData['FinSearch.config.crossSellingCategories'].length
+        ? await this.categoryRepository.search(this.selectedCategoriesCriteria, Shopware.Context.api)
+        : new EntityCollection(
+          this.categoryRepository.route,
+          this.categoryRepository.entityName,
+          Shopware.Context.api,
+          null,
+          []
```

Comment:
`criteria` and `entities` already should have sufficient defaults.

![image](https://user-images.githubusercontent.com/20555112/159239589-7c7a1ad2-b17c-4257-9925-daada8e6ed31.png)